### PR TITLE
Refactor eto modules

### PR DESCRIPTION
**Pull request by trival**
see https://github.com/Neufund/platform-frontend/issues/1309

### CODE REVIEW

**Comment by Moejoe90** [comment]
@trival do you see this critical before the release?

File: app/modules/public-etos/sagas.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    );
+  } catch (e) {
+    notificationCenter.error("Could load ETO preview. Is the preview link correct?");
+    yield put(actions.routing.goToDashboard());
+  }
+}
+
+function* loadEtos({ apiEtoService, logger }: TGlobalDependencies): any {
+  try {
+    const etoData: IHttpResponse<TPublicEtoData[]> = yield apiEtoService.getEtos();
```
Comment (desfero) [inline app/modules/public-etos/sagas.ts]: rename variable to `etosResponse` as it contain more than single eto

File: app/modules/public-etos/sagas.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    const etos: { [id: string]: TPublicEtoData } = {};
```
Comment (desfero) [inline app/modules/public-etos/sagas.ts]: Can be done without mutations

```
const etos = keyBy(etosResponse.body, eto => eto.etoId);
const order = Object.keys(etos)
```

`keyBy` is from `lodash`

File: app/modules/public-etos/sagas.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      if (eto.etoId) {
+        etos[eto.etoId] = eto;
+        order.push(eto.etoId);
+      }
+    });
+
+    yield put(actions.publicEtos.setPublicEtos(etos));
+    yield put(actions.publicEtos.setEtosDisplayOrder(order));
+  } catch (e) {
+    logger.error("ETOs could not be loaded", e, e.message);
```
Comment (desfero) [inline app/modules/public-etos/sagas.ts]: do we need to add `e.message` as it's already in `e`?

File: app/modules/investmentFlow/sagas.ts [hunk 4]
```diff
@@ -233,12 +226,17 @@ function* recalculateCurrencies(): any {
   }
 }
 
+function* cancelInvestment(): any {
+  yield put(actions.txSender.txSenderHideModal());
```
[the remark this answers is not in the document]
Comment (trival) [reply app/modules/investmentFlow/sagas.ts]: lets look at that later, and refactor general tx flow. (txSenderHideModal cancels the whole flow there)

File: app/components/walletSelector/walletRecover/__tests__/WalletRecover.spec.e2e.ts [hunk 5]
```diff
@@ -39,7 +39,7 @@ describe("Wallet recover", () => {
 
   const expectedGeneratedAddress = "0x429123b08DF32b0006fd1F3b0Ef893A8993802f3";
 
-  it.only("should recover wallet from saved phrases", () => {
+  it("should recover wallet from saved phrases", () => {
```
Comment (Moejoe90) [inline app/components/walletSelector/walletRecover/__tests__/WalletRecover.spec.e2e.ts]: :+1:

**Pull request merged**